portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -31,19 +38,32 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
     Gson gson = new Gson();
-    response.setContentType("text/html;");
-    response.getWriter().println(gson.toJson(comments));
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(results.asList(FetchOptions.Builder.withLimit(10))));
   }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-      String comment = request.getParameter("comment");
-      if (comment == null) {
+      String name = request.getParameter("fname");
+      String comment = request.getParameter("comment");  
+      long timestamp = System.currentTimeMillis();    
+
+      Entity commentEntity = new Entity("Comment");
+      commentEntity.setProperty("name", name);
+      commentEntity.setProperty("comment", comment);
+      commentEntity.setProperty("timestamp", timestamp);
+      
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+      if (comment == null || name == null) {
         response.sendError(400);
         return;
       } else if (!comment.isEmpty()) {
```

Comment:
When would the comment be empty but not null? Do you need this if statement?

Reply:
It would be null if there is no comment or name HTML element and empty if nothing is entered when the users presses submit. Hao wanted me to account for this in my last PR.

Reply:
In this case maybe instead of silently ignoring this comment, display an error message on the webpage to tell the user that empty comments are not allowed? 

Feel free to do so in a follow up PR, though!